File: app/src/main/java/com/ninhttd/devtest/custom/LogUtils.java
package com.ninhttd.devtest.custom;

import android.util.Log;

import com.ninhttd.devtest.BuildConfig;

import org.apache.commons.lang3.StringUtils;

public class LogUtils {
    public static final String PREFIX_TAG = "PRISM_";
    public static boolean isDebugLogEnable = BuildConfig.DEBUG;

    public static void debugLog(final Object target, final String content) {
        if (isDebugLogEnable) {
            Log.d(PREFIX_TAG + target.getClass().getSimpleName(),
                    StringUtils.isEmpty(content) ? "[Log is empty]" : content);
        }
    }

    public static void errorLog(final Object target, final String content) {
        Log.i(PREFIX_TAG + target.getClass().getSimpleName(), StringUtils.isEmpty(content) ? "[Log is empty]" : content);
    }
}

File: app/src/main/java/com/ninhttd/devtest/custom/LogUtils.kt
package com.ninhttd.devtest.custom

import android.util.Log

import com.ninhttd.devtest.BuildConfig

import org.apache.commons.lang3.StringUtils

object LogUtils {
    val PREFIX_TAG = "PRISM_"
    var isDebugLogEnable = BuildConfig.DEBUG

    fun debugLog(target: Any, content: String) {
        if (isDebugLogEnable) {
            Log.d(PREFIX_TAG + target.javaClass.simpleName,
                    if (StringUtils.isEmpty(content)) "[Log is empty]" else content)
        }
    }

    fun errorLog(target: Any, content: String) {
        Log.i(PREFIX_TAG + target.javaClass.simpleName, if (StringUtils.isEmpty(content)) "[Log is empty]" else content)
    }
}
